src/components/configure/header/Header.tsx

```diff
@@ -97,6 +110,159 @@ export default class Header extends React.Component<HeaderProps, HeaderState> {
     this.setState({ logoAnimation: true });
   }
 
+  handleAuthMenuIconClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    this.setState({
+      authMenuAnchorEl: event.currentTarget,
+    });
+  };
+
+  handleAuthMenuClose = () => {
+    this.setState({
+      authMenuAnchorEl: null,
+    });
+  };
+
+  async handleLogoutMenuClick() {
+    this.props.logout!();
+    this.setState({
+      authMenuAnchorEl: null,
+    });
+  }
+
+  async handleLoginMenuClick() {
+    this.setState({ authMenuAnchorEl: null, openAuthProviderDialog: true });
+  }
+
+  async handleLinkGoogleAccountMenuClick() {
+    this.setState({ authMenuAnchorEl: null });
+    this.props.linkToGoogleAccount!();
+  }
+
+  async handleLinkGitHubAccountMenuClick() {
+    this.setState({ authMenuAnchorEl: null });
+    this.props.linkToGitHubAccount!();
+  }
+
+  renderLinkGoogleAccountMenu() {
```

Comment:
FYI: JS's top-revel function which return jsx is useful in this case.
I believe a good sample is as below:
https://github.com/remap-keys/remap/blob/main/src/components/configure/keymap/Keymap.tsx#L306-L328

Reply:
@adamrocker I could not decide which creating top-level function or creating render method is better until now. Ok, I intend to create such top-level function from next opportunity.